AmethystAPI/src/amethyst/runtime/patchManager/PatchManager.cpp

```diff
@@ -0,0 +1,67 @@
+#include "PatchManager.h"
+#include "../../Log.h"
+#include "../../Memory.h"
+namespace Amethyst {
+    // Returns false if the patch failed to apply
+    bool PatchManager::ApplyPatch(uintptr_t address, uint8_t* patch, size_t size){
+        DWORD oldProtection;
+        if (!UnprotectMemory(address, size, &oldProtection)) {
+            Assert("Failed to unprotect memory at address: 0x{0:x}", address);
+			return false;
+		}
+
+        uint64_t boundsLow = address;
+        uint64_t boundsHigh = address + size;
+        for (auto& establishedPatch : this->mPatches) {
+			if (establishedPatch.address == address) {
+                Log::Warning("Patch already exists at address: 0x{0:x}", address);
+			}
+            uint64_t establishedBoundsLow = establishedPatch.address;
+            uint64_t establishedBoundsHigh = establishedPatch.address + establishedPatch.original.size;
+            if (boundsLow >= establishedBoundsLow && boundsLow <= establishedBoundsHigh
+                || boundsHigh >= establishedBoundsLow && boundsHigh <= establishedBoundsHigh) {
+                Log::Warning("Patch at address: 0x{0:x} overlaps with existing patch at address: 0x{1:x}", address, establishedPatch.address);
+			}
+		}
+        Patch ptch;
+		OriginalMemory original;
+		uint8_t* originalMemory = (uint8_t*)malloc(size);
+        if (!originalMemory) 
+			Assert("Failed to allocate memory for original bytes at address: 0x{0:x}", address);
+        memcpy(originalMemory, reinterpret_cast<void*>(address), size);
+		original.original = originalMemory;
+        original.size = size;
+		ptch.original = original;
```

Comment:
There seems to be a few formatting quirks here. It might just be the git website interface?

Reply:
Yeah that and my visual studio really doesnt like this project but mostly github

Reply:
On my end it looks fine 
![image](https://github.com/FrederoxDev/Amethyst/assets/107327509/6055cade-9b22-4cc9-97f6-b5f7cd49db36)